src/Voter.jsx

```diff
@@ -0,0 +1,38 @@
+import { addVote } from './Api'
+import { useEffect, useState } from "react"
+import { Link } from 'react-router-dom';
+
+
+function Voter({article_id, currentVotes, setVotes}) {
+const [hasVoted, setHasVoted] = useState(false);
+
+
+const updateVote = () => {
+let vote = 1    
+if (hasVoted) {vote = 0}
+setVotes(vote)
+addVote(article_id, vote? 1 : -1)
+setHasVoted((hasVoted) => {
+return !hasVoted
```

Comment:
I really like this `hasVoted` state you have to undo a user's vote if they have already voted!
If you set `vote = -1` on _line 12_ then you could remove the ternary on _line 14_ 💡